src/redux/sagas/rootSaga.js

```diff
@@ -0,0 +1,19 @@
+import { put, takeLatest } from "redux-saga/effects";
+
+import { receiveIncrement } from "../actions/articleActions";
+import { REQUEST_INCREMENT } from "../actions/actionTypes";
+
+// worker Saga: will be fired once user request actions
+function* increment(action) {
+  try {
+    // do an api call here
+    yield put(receiveIncrement());
+  } catch (e) {
+    console.log(e);
+  }
+}
+
+// once the REQUEST_INCREMENT action type is called, increment method is run
+export default function* saga() {
```

Comment:
This is the `watcher` which should be yielded amongst other `watchers` in the `rootSaga`, that we should create as pointed out earlier. It's work is to listen for `actions` once it catches an `action`, it instructs the slave `saga` it has registered in `takeLatest` below, in this case the `incrementSaga`, handing it the action for processing(*as you saw, the `incrementSaga` takes `action` as argument*).

I hope the flow is getting clearer.